contracts/bonding/BondingManager.sol

```diff
@@ -498,41 +529,133 @@ contract BondingManager is IBondingManager, Controllable {
+        Delegator storage del = delegators[_delegator];
+
+        if (del.withdrawRound > 0) {
+            // Delegator called unbond
+            if (roundsManager().currentRound() >= del.withdrawRound) {
+                return DelegatorStatus.Unbonded;
+            } else {
+                return DelegatorStatus.Unbonding;
+            }
+        } else if (del.transcoderAddress != address(0) && transcoders[del.transcoderAddress].withdrawRound > 0) {
+            // Transcoder resigned
+            if (roundsManager().currentRound() >= transcoders[del.transcoderAddress].withdrawRound) {
+                return DelegatorStatus.Unbonded;
+            } else {
+                return DelegatorStatus.Unbonding;
+            }
+        } else if (del.startRound > roundsManager().currentRound()) {
+            // Delegator round start is in the future
+            return DelegatorStatus.Pending;
+        } else if (del.startRound <= roundsManager().currentRound()) {
+            // Delegator round start is now or in the past
+            return DelegatorStatus.Bonded;
+        } else {
+            // Delegator in inactive phase
+            return DelegatorStatus.Inactive;
+        }
+    }
 
-        return delegators[_delegator].bondedAmount.add(delegatorRewards(_delegator));
+    /*
+     * @dev Return number of minted tokens for a reward call
+     */
+    function mintedTokensPerReward() public constant returns (uint256) {
+        return initialTokenSupply.mul(initialYearlyInflation).div(100).div(roundsManager().rewardCallsPerYear());
     }
 
     /*
-     * @dev Computes rewards for a delegator since last state transition
+     * @dev Computes token distribution for delegator since its last state transition
      * @param _delegator Address of delegator
      */
-    function delegatorRewards(address _delegator) public constant returns (uint256) {
-        uint256 rewards = 0;
+    function delegatorTokenPoolsShare(Delegator storage del) internal constant returns (uint256) {
+        uint256 tokens = 0;
 
-        // Check if delegator bonded to a transcoder
-        if (delegators[_delegator].transcoderAddress != address(0)) {
+        if (del.transcoderAddress != address(0)) {
             // Iterate from round that delegator last transitioned states to current round
-            // If the delegator is bonded to a transcoder, it has been bonded to the transcoder since lastStateTransitionRound
-            for (uint256 i = delegators[_delegator].lastStateTransitionRound; i <= roundsManager().currentRound(); i++) {
-                uint256[2] rewardMultiplier = rewardMultiplierPerTranscoderAndRound[delegators[_delegator].transcoderAddress][i];
+            // If the delegator is bonded to a transcoder, it has been bonded to the transcoder since lastStakeUpdateRound
+            for (uint256 i = del.lastStakeUpdateRound; i <= roundsManager().currentRound(); i++) {
```

Comment:
Note for the future that we'll probably want to use the loop limit pattern where this executes the min of # of rounds since last state transition, and some safe loop param like 100. And then it will need to be called multiple times to get current. I'll file an issue.